### PR TITLE
🔒 Upgrade plugins for Jenkins advisory 2018-06-04

### PR DESCRIPTION
**Pull request by grdryn**
Advisory: https://jenkins.io/security/advisory/2018-06-04/

kubernetes 1.6.2 -> 1.7.1
https://github.com/jenkinsci/kubernetes-plugin/blob/master/CHANGELOG.md

git 3.8.0 -> 3.9.1
https://plugins.jenkins.io/git

github 1.26.0 -> 1.29.1
https://plugins.jenkins.io/github

github-branch-source 2.2.3 -> 2.3.6
https://plugins.jenkins.io/github-branch-source

### CODE REVIEW

File: 2/contrib/openshift/base-plugins.txt [hunk 1]
```diff
@@ -7,8 +7,9 @@ openshift-client:1.0.12
 # 1.6.0 has the support we added for allowing the use of pod yaml for pod template
 # config from the config panel (there was a flavor for this already with the k8s plugin's
 # pipeline DSL, and 1.6.2 has the support for opting in the transfer of proxy related
-# env vars from the master to the client
-kubernetes:1.6.2
+# env vars from the master to the client, and 1.7.1 fixes
+# https://issues.jenkins-ci.org/browse/SECURITY-883
```
Comment (grdryn) [inline 2/contrib/openshift/base-plugins.txt]: I just updated this comment in the most basic way possible. Let me know if you'd like me to change it.

**Comment by gabemontero** [comment]
Hmmmm .... more jdk package validation pain, this time with the centos build, like we saw not too long ago with the rhel builds:

```
Complete!
.M.......  g /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.171-8.b10.el7_5.x86_64/jre/lib/amd64/server/classes.jsa
.M.......  g /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.171-8.b10.el7_5.i386/jre/lib/i386/server/classes.jsa
.M.......  g /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.171-8.b10.el7_5.i386/jre/lib/i386/server/classes.jsa
The command '/bin/sh -c yum install -y centos-release-scl-rh &&     x86_EXTRA_RPMS=$(if [ "$(uname -m)" == "x86_64" ]; then echo -n java-1.8.0-openjdk-headless.i686 ; fi) &&     INSTALL_PKGS="bc gettext git java-1.8.0-openjdk-headless lsof rsync tar unzip which zip bzip2" &&     yum install -y --setopt=tsflags=nodocs install $INSTALL_PKGS $x86_EXTRA_RPMS &&     rpm -V $INSTALL_PKGS $x86_EXTRA_RPMS &&     yum clean all &&     mkdir -p /home/jenkins &&     chown -R 1001:0 /home/jenkins &&     chmod -R g+w /home/jenkins &&     chmod 664 /etc/passwd &&     chmod -R 775 /etc/alternatives &&     chmod -R 775 /var/lib/alternatives &&     chmod -R 775 /usr/lib/jvm &&     chmod 775 /usr/bin &&     chmod 775 /usr/lib/jvm-exports &&     chmod 775 /usr/share/man/man1 &&     chmod 775 /var/lib/origin &&     unlink /usr/bin/java &&     unlink /usr/bin/jjs &&     unlink /usr/bin/keytool &&     unlink /usr/bin/orbd &&     unlink /usr/bin/pack200 &&     unlink /usr/bin/policytool &&     unlink /usr/bin/rmid &&     unlink /usr/bin/rmiregistry &&     unlink /usr/bin/servertool &&     unlink /usr/bin/tnameserv &&     unlink /usr/bin/unpack200 &&     unlink /usr/lib/jvm-exports/jre &&     unlink /usr/share/man/man1/java.1.gz &&     unlink /usr/share/man/man1/jjs.1.gz &&     unlink /usr/share/man/man1/keytool.1.gz &&     unlink /usr/share/man/man1/orbd.1.gz &&     unlink /usr/share/man/man1/pack200.1.gz &&     unlink /usr/share/man/man1/policytool.1.gz &&     unlink /usr/share/man/man1/rmid.1.gz &&     unlink /usr/share/man/man1/rmiregistry.1.gz &&     unlink /usr/share/man/man1/servertool.1.gz &&     unlink /usr/share/man/man1/tnameserv.1.gz &&     unlink /usr/share/man/man1/unpack200.1.gz' returned a non-zero code: 2
make: *** [build] Error 2
++ export status=FAILURE
++ status=FAILURE
```

@bparees @openshift/sig-continuous-delivery FYI

Assuming this is happening consistently, we'll most likely have to employ a similar workaround wrt the centos build's use of rpm -V

@grdryn - I'll handle that in a separate PR if it comes to that.

**Comment by gabemontero** [comment]
yep it is consistent

it will probably be tomorrow but I'll get a PR together that mimics what we did wrt rhel and the rpm -V call for the jdk's and we'll see what happens.

**Comment by gabemontero** [comment]
fyi ... the build passes now ... it is in the middle of the tests

**Comment by gabemontero** [comment]
Installed plugins:
```
ace-editor:1.0.1
apache-httpcomponents-client-4-api:4.5.3-2.0
authentication-tokens:1.1
blueocean-autofavorite:1.2.0
blueocean-bitbucket-pipeline:1.4.1
blueocean-commons:1.4.1
blueocean-config:1.4.1
blueocean-core-js:1.4.1
blueocean-dashboard:1.4.1
blueocean-display-url:2.1.1
blueocean-events:1.4.1
blueocean-github-pipeline:1.4.1
blueocean-git-pipeline:1.4.1
blueocean-i18n:1.4.1
blueocean-jira:1.4.1
blueocean:1.4.1
blueocean-jwt:1.4.1
blueocean-personalization:1.4.1
blueocean-pipeline-api-impl:1.4.1
blueocean-pipeline-editor:1.4.1
blueocean-pipeline-scm-api:1.4.1
blueocean-rest-impl:1.4.1
blueocean-rest:1.4.1
blueocean-web:1.4.1
branch-api:2.0.11
cloudbees-bitbucket-branch-source:2.2.5
cloudbees-folder:6.1.2
conditional-buildstep:1.3.1
config-file-provider:2.17
credentials-binding:1.15
credentials:2.1.15
display-url-api:2.1.0
docker-commons:1.11
docker-workflow:1.9
durable-task:1.16
favorite:2.3.1
git-client:2.7.0
github-api:1.90
github-branch-source:2.3.6
github:1.29.1
github-organization-folder:1.6
git:3.9.1
git-server:1.5
handlebars:1.1
htmlpublisher:1.16
jackson2-api:2.7.3
jenkins-design-language:1.4.1
jira:2.4.2
job-dsl:1.68
jquery-detached:1.2.1
jsch:0.1.54.1
junit:1.24
kubernetes-credentials:0.3.0
kubernetes:1.7.1
mailer:1.21
mapdb-api:1.0.9.0
matrix-auth:2.2
matrix-project:1.12
mercurial:2.3
momentjs:1.1
openshift-client:1.0.12
openshift-login:1.0.8
openshift-pipeline:1.0.54
openshift-sync:1.0.19
parameterized-trigger:2.35.2
pipeline-build-step:2.7
pipeline-github-lib:1.0
pipeline-graph-analysis:1.6
pipeline-input-step:2.8
pipeline-milestone-step:1.3.1
pipeline-model-api:1.2.5
pipeline-model-declarative-agent:1.1.1
pipeline-model-definition:1.2.5
pipeline-model-extensions:1.2.5
pipeline-rest-api:2.9
pipeline-stage-step:2.2
pipeline-stage-tags-metadata:1.2.5
pipeline-stage-view:2.4
pipeline-utility-steps:1.5.1
plain-credentials:1.3
pubsub-light:1.12
run-condition:0.10
scm-api:2.2.6
script-security:1.41
sse-gateway:1.15
ssh-credentials:1.13
structs:1.13
subversion:2.10.3
token-macro:2.0
variant:1.1
workflow-aggregator:2.5
workflow-api:2.25
workflow-basic-steps:2.3
workflow-cps-global-lib:2.7
workflow-cps:2.42
workflow-durable-task-step:2.18
workflow-job:2.14.1
workflow-multibranch:2.16
workflow-remote-loader:1.4
workflow-scm-step:2.4
workflow-step-api:2.14
workflow-support:2.18
```

**Comment by gabemontero** [comment]
will create rhel plugin rpm when 3.11 builds are running in dist git

**Pull request merged**